### ISSUE TITLE
Automatically detect the saddle point problem in BlockOperator

### ISSUE THREAD

**Issue by tangqi** [mfem/mfem#3959]
Hi,
I have a 2x2 block system that is singular in the two diagonal blocks. Therefore, I want to form a different index set of this 2x2 block system based on if the diagonal is 0 or not. I can get the diagonal vectors of two diagonal blocks and thus get the needed index sets. But I am not sure if there is an easy way to get all the sub-matrices and form a new 2x2 system. 

I think I want to `GetSubMatrix` but remain a SparseMatrix and eventually glue things together. Is there an easy way to implement that, at least for a serial case?

Basically, I want to realize `-pc_fieldsplit_detect_saddle_point` inside mfem. Alternatively, I can use petsc but I am not sure it will work on the block system through the current mfem's petsc interface, which gives `MATNEST` instead of `MATAIJ`. Not sure one can call `-pc_fieldsplit_detect_saddle_point` on a nest matrix whose index set has already been determined through the nest structure. 

Qi

**Comment by tangqi** [comment]
close for now. there might be another workaround.
